content/code/markdown.md

```diff
@@ -2,14 +2,31 @@
 
 ## General
 
+* Always keep the scope of the document in mind. A document should precisely fulfill its purpose, nothing more, nothing less.
+  It is a common pitfall to end up going into rabbit holes and spending half a document explaining irrelevant details.
+* Try to write short sentences whenever possible to avoid complex grammar, complex use of tenses, ambiguous pronouns and so on.
+  A good guideline when it comes to technical writing is to aim for 20-30 words per sentence.
+  Keeping sentences short should however never come at the expense of clarity, syntactic cues and important information.
+* Just like in code, consistency is key.
+
+### Sections
+
 * Section titles should follow the [Chicago Title Capitalization](https://en.wikipedia.org/wiki/Title_case#Chicago_Manual_of_Style) standard.
-* Lists should use the `*` character rather than the `-` character.
 * Documents should start with a level one heading and should ideally be the same as the file name.
-* For long files that are not served within this repository, it is best to have a table of contents at the end of the introduction of the level one heading section.
+* Sections should be ordered hierarchically. Each document starts with a level one heading (`#`), which contains one or more level two headings (`##`), which contain one or more level threes (`###`) and so on.
+
+### Formatting
+
+* Lists should use the `*` character rather than the `-` character.
 * Keep extra newlines between paragraphs and sections to make the source easier to read.
+* Paragraphs that include multiple sentences should have the sentences on separate lines, so that updating one sentence results in a clear diff where one line changes.
+* For long files that are not served within this repository, it is best to have a table of contents at the end of the introduction of the level one heading section.
```

Comment:
Maybe we can also link to the specific sections in the table of contents itself so that one can click and easily reach the sections?

I am not sure exactly what `that are not served within this repository` means in the context of table of contents.

Reply:
It means for example, a markdown documentation file in a code repository. This particular repo has no need for tables of contents, since it uses mkdocs which automatically generates a navigation sidebar, but our usual Markdown files are accessed through GitHub which currently doesn't have this feature. Because of that, adding our own ToC can be a nice solution for large documents to be more accessible. It's also neat for some IDE users.